uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1004.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+trait CRM_HRCore_Upgrader_Steps_1004 {
+
+  /**
+   * Upgrader to set 'Home' as default location and set all 'Home' addresses as
+   * primary.
+   *
+   * @return bool
+   */
+  public function upgrade_1004() {
+    $this->up1004_setHomeLocationAsDefault();
+    $this->up1004_setAllHomeAddressesAsPrimary();
+
+    return true;
+  }
+
+  private function up1004_setHomeLocationAsDefault() {
+    civicrm_api3('LocationType', 'get', [
+      'sequential' => 1,
+      'name' => 'Home',
+      'api.LocationType.create' => [
+        'id' => '$value.id',
+        'is_default' => 1,
+        'is_reserved' => 1,
```

Comment:
I asked Mike about this and suggested we share it with the rest of the CiviHR team, asking if we should make 'Home' reserved, or change the SSP to use primary address instead. The decision was to keep SSP hard-wired to the 'Home' address location type and make it reserved, as this was the original intention behind the functionality on the SSP (allow users to check and update their current home address).